docs/DeveloperGuide.md

```diff
@@ -4,30 +4,148 @@
+ - `ParserAdd#substringIndex(type)` -- Returns the substring index of the start of the user input.
+
+The `CommandAdd#execute()` operation is exposed in the main `Duke` class, while the operations in `ParserAdd` are
+exposed in the `Parser` class.
+
+Given below is an example usage of the feature.
+![](./diagrams/AddFeature.png)
+
+Step 1. The user executes `add amt/24 t/02-02-2012` as the `userInput` to add a new `Expense` into the the list of 
+expenses. `Duke` calls `Parser#extractCommandKeyword(userInput)` to parse the input and determines that the `add` command is called.
+
+Step 2. `Duke` instantiates a new `CommandAdd` and calls `CommandAdd#execute()`, which in turn calls 
+`Parser#extractAddParameters(userInput)` and `ExpenseList#getExpenseList()`.
+
+Step 3. `Parser#extractAddParameters(userInput)` then calls `ParserAdd#parseInput(userInput)` and returns the parsed 
+input as a string of words `parsedInput[]`.
+
+Step 4. `CommandAdd#execute` instantiates a new `Expense` object with the returned `parsedInput[]` and adds it to
+`expenseList`.
+
+### "Delete" feature
+
+This mechanism is facilitated by `CommandDelete`, which extends `Command`. It makes use of the output from `Parser`,
+which takes in the user input as string and returns the index of the expense to be deleted, and the `CommandDelete`
+proceeds to remove the respective expense from the list of expenses stored in `expenseList`.
+
+`CommandDelete` implements the following operation:
+- `CommandDelete#execute()` -- Removes expense based on the index specified by user.
+
+`Parser` implements the following operation:
+- `Parser#extractIndex(userInput)` -- Extracts the index of the expense to be deleted, as specified by the user.
+
+The `CommandDelete#execute()` operation is exposed in the main `Duke` class. The `Parser#extractIndex(userInput)`
+is exposed in the `parser` class.
+
+Given below is an example usage scenario and how the 'delete' mechanism behaves at each step.
+
+![](./diagrams/DeleteFeature.png)
+
+Step 1. The user executes `delete 1` command to delete the 1st expense in the expense list. `Duke#run()` calls
+`Parser#extractCommandKeyword(userInput)` to parse the input and determine that the `delete` command is called.
+
+Step 2. `Duke` instantiates a new `CommandDelete` and calls `CommandDelete#execute()`, which then calls
+`expenseList.getExpenseList()` and `parser.extractIndex(input)`.
+
+Step 3. `CommandDelete#execute()` removes the expense at index specified by the user.
+
+### Monthly Overview
+
+This mechanism is facilitated by `CommandOverview`, which extends `Command`. It makes use of output from `Parser`
+to extract `month` and `year` from user input. It then calls on `Monthly Overview` if both `month` and 'year' 
+are not null, which makes use of `MonthFilter` to filter out expenses in that specific month and returns sum by 
+category sorted in descending order before printing out the final overview in the intended format.
+
```

Comment:
Formatting error with `month` and 'year', I think can consider describing / explaining how YearFilter works besides just MonthFilter as both methods are called.

For the diagram I think there's a small typo for filteredExpenses(), otherwise good job, it looks quite detailed and readable.